railties/lib/rails/engine.rb

```diff
@@ -618,6 +618,13 @@ def load_seed
       end
     end
 
+    initializer :add_mailer_preview_paths do
+      previews = paths["test/mailers/previews"].existent
+      unless previews.empty?
```

Comment:
The Action Mailer railtie doesn't add these paths if `show_previews` is false. Should we do the same here?

Is the `empty?` check purely an optimisation? If so, maybe it makes sense to add it in the railtie, too.

Reply:
> The Action Mailer railtie doesn't add these paths if show_previews is false. Should we do the same here?

I think this is not necessary. We are not forced to keep `preview_paths` empty when `show_previews == false`, because previews just won't be showed in `/rails/mailers` by that setting. But avoiding that check (which should be in `on_load` block due to that we need `ActionMailer` loaded at that time) we simplified code a bit.

> Is the empty? check purely an optimisation? If so, maybe it makes sense to add it in the railtie, too.

Yes, it is some sort of unnecessary optimization, similarly as in `:add_routing_paths` and `:add_view_paths` initializers above. I think this is also not needed for railtie (to simplify code a bit, like in previous comment).

Reply:
Good point about the `on_load` hook - it won't even work in development, since it doesn't fire until the first time a mailer is used, at which point the autoload paths are already frozen.

I'd like to avoid the situation where `preview_paths` contains paths from engines but not from the application, so I think we should take the `show_previews` check out of the railtie instead.